libs/designsystem/src/lib/components/page-local-navigation/page-local-navigation.component.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { WindowRef } from '../../types';
+import { LocalNavigationItem } from './page-local-navigation-item';
+
+@Component({
+  selector: 'kirby-page-local-navigation',
+  templateUrl: './page-local-navigation.component.html',
+  styleUrls: ['./page-local-navigation.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PageLocalNavigationComponent implements AfterViewInit {
+  private readonly DEBOUNCE_TIME_MS = 250;
+
+  selectedIndex$ = new BehaviorSubject(0);
+  private _selectedIndex = 0;
+  @Input() set selectedIndex(value: number) {
+    if (value > -1 && value !== this._selectedIndex) {
+      this._selectedIndex = value;
+      this.selectedIndex$.next(value);
+      this.scrollToSelectedTab(value);
+    }
+  }
+  get selectedIndex(): number {
+    return this._selectedIndex;
+  }
+
```

Comment:
Decide to use _selectedIndex or selectedIndex$ not both. 
If choosing the latter you may subscribe to it in order to update the selected tab - simplifying the busy setter method.